sdks/java/extensions/zetasketch/src/main/java/org/apache/beam/sdk/extensions/zetasketch/HllCountInitFn.java

```diff
@@ -0,0 +1,153 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.beam.sdk.extensions.zetasketch;
+
+import static org.apache.beam.vendor.guava.v26_0_jre.com.google.common.base.Preconditions.checkArgument;
+
+import com.google.zetasketch.HyperLogLogPlusPlus;
+import com.google.zetasketch.shaded.com.google.protobuf.ByteString;
+import org.apache.beam.sdk.coders.Coder;
+import org.apache.beam.sdk.coders.CoderRegistry;
+import org.apache.beam.sdk.transforms.Combine;
+
+/**
+ * {@link Combine.CombineFn} for the {@link HllCount.Init} combiner.
+ *
+ * @param <InputT> type of input values to the function
+ * @param <HllT> type of the HLL++ sketch to compute
+ */
+abstract class HllCountInitFn<InputT, HllT>
+    extends Combine.CombineFn<InputT, HyperLogLogPlusPlus<HllT>, byte[]> {
+
+  private int precision;
+
+  private HllCountInitFn() {
+    setPrecision(HllCount.DEFAULT_PRECISION);
+  }
+
+  int getPrecision() {
+    return precision;
+  }
+
+  // Should not be called at pipeline execution time.
```

Comment:
Would expand on this (e.g., what happens if the advice is ignored?), and also turn it into Javadoc: 
"this will only have an effect if called before ...; any updates after that will be discarded; only intended to be used from within a builder" or similar.

Reply:
> Would expand on this (e.g., what happens if the advice is ignored?), and also turn it into Javadoc: "this will only have an effect if called before ...; any updates after that will be discarded; only intended to be used from within a builder" or similar.

Done! PTAL.

> Since you have a builder in the caller class, I was wondering whether there is a good way to only allow setting the precision at construction time, which would be cleaner. But I think it would require a enum for the allowed types in the builder... (I'll think about it some more).

Exactly! I think we have went through the same thought process. (In an earlier version of the code, the `precision` field is a `final` field, but after a few rounds of refactoring we need to call it from within a builder.) To make the coder cleaner, I did not add the extra enum, but created this package-private function.